lib/src/import_path_base.dart

```diff
@@ -0,0 +1,276 @@
+  late String searchRoot;
+
+  ImportPath(this.from, Object importToFind,
+      {this.findAll = false,
+      bool quiet = false,
+      this.strip = false,
+      this.fastParser = false,
+      this.includeConditionalImports = true,
+      String? searchRoot,
+      MessagePrinter messagePrinter = print})
+      : importToFind = ImportToFind.from(importToFind),
+        super(quiet: quiet, messagePrinter: messagePrinter) {
+    this.searchRoot = searchRoot ?? resolveSearchRoot();
+  }
+
+  /// This list contains common Dart root directories.
+  /// These names are preserved by [resolveSearchRoot] to prevent
+  /// them from being stripped. Default: `'web', 'bin', 'src', 'test', 'example'`
+  List<String> commonRootDirectories = ['web', 'bin', 'src', 'test', 'example'];
+
+  /// Resolves the [searchRoot] using [from] parent.
+  /// See [commonRootDirectories].
+  String resolveSearchRoot() {
+    var rootPath = p.dirname(from.path);
+    var rootDirName = p.split(rootPath).last;
+
+    if (commonRootDirectories.contains(rootDirName)) {
+      var rootPath2 = p.dirname(rootPath);
+      if (rootPath2.isNotEmpty) {
+        rootPath = rootPath2;
+      }
+    }
+
+    var rootUri = from.replace(path: rootPath).toString();
+    return rootUri.endsWith('/') ? rootUri : '$rootUri/';
+  }
+
+  /// Return the search root to [strip] from the displayed import paths.
+  /// If [strip] is `false` returns `null`.
+  /// See [searchRoot].
+  String? get _stripSearchRoot => strip ? searchRoot : null;
+
+  /// Executes the import search and prints the results.
+  /// - [style] defines the output format. Default: [ImportPathStyle.elegant]
+  /// - See [search], [printMessage] and [ASCIIArtTree].
+  Future<ASCIIArtTree?> execute(
+      {ImportPathStyle style = ImportPathStyle.elegant,
+      Directory? packageDirectory}) async {
+    var tree = await search(
+        style: style.asASCIIArtTreeStyle ?? ASCIIArtTreeStyle.elegant,
+        packageDirectory: packageDirectory);
+
+    if (tree != null) {
+      if (!quiet) {
+        printMessage('');
+      }
+
+      if (style == ImportPathStyle.json) {
+        var j = JsonEncoder.withIndent('  ').convert(tree.toJson());
+        printMessage(j);
+      } else {
+        var treeText = tree.generate();
+        printMessage(treeText);
+      }
+    }
+
+    if (!quiet) {
+      if (tree == null) {
+        printMessage(
+            '» Unable to find an import path from $from to $importToFind');
+      } else {
+        var totalFoundPaths = tree.totalLeaves;
+        if (totalFoundPaths > 1) {
+          printMessage(
+              '» Found $totalFoundPaths import paths from $from to $importToFind\n');
+        }
+      }
+    }
+
+    return tree;
+  }
+
+  /// Performs the imports search and returns the tree.
+  /// - [style] defines [ASCIIArtTree] style. Default: [ASCIIArtTreeStyle.elegant]
+  /// - See [searchPaths] and [ASCIIArtTree].
+  Future<ASCIIArtTree?> search(
```

Comment:
I don't think this method should be on this class - at most it could be an extension method? I don't think its something that should really be overridden etc, and it's tightly coupling this class to the ASCIIArtTree itself which isn't a good separation of concerns. 

I would probably rename `searchPaths` to just `search`, and drop this (its only used in one place and trivial to inline there).